### PR TITLE
Remove propTypes assignment entirely by default

### PR DESCRIPTION
**Pull request by billyjanitsch**
Currently, when a production bundle is built, prop types assignment is transpiled & minified to:

```js
Foo.propTypes = {}
```

With this change, that statement is removed entirely, saving a few bytes in each component.

The downside is that it's a slightly less safe transform because accessing a specific prop type (e.g. `Foo.propTypes.className`) will throw in production instead of evaluating to `undefined`. IMO this is fine because we should never be doing this, and if we ever do need to, we can export the `propTypes` object manually which de-opts the transform.

Note that something like `{...Foo.propTypes, bar: PropTypes.string}` (which does occasionally appear in our codebase, for better or worse) will not throw regardless.

~(Please hold off on merging for now because this is a breaking change and I'd like to get a minor release out first.)~ Nvm, going to release this + #57 + #58 in a major anyway.

### CODE REVIEW

**Comment by jhhayashi** [comment]
do you know how much this will reduce our bundle size? it seems like there's decent risk (crashing our app), and i'm not sure that saving a few btyes per component would be worth the risk.

on the other hand, if the goal is to also use this to strictly enforce that we never access a prop type directly (and shaving a little off the bundle size would just be a bonus), then i'd be happy to roll this out

**Comment by billyjanitsch** [comment]
>it seems like there's decent risk (crashing our app)

We already have this plugin set to `'remove'` when the Babel env is `'production'` (true for all apps we have deployed) which is even more extreme than `'unsafe-wrap'`. (CRA has the [same default](https://github.com/facebook/create-react-app/blob/11580f22f23758b5aed85431a389a78b40d64927/packages/babel-preset-react-app/create.js#L116-L119), FWIW.)

IMO the risk is extremely low, because there's no practical use case for accessing a specific prop type. If it helps reassure you, there are no instances of `/propTypes\./` anywhere in our codebase.

This change would only affect our library builds, where I consider the risk even lower since it's even less likely to access `propTypes` across a library boundary. It's essentially aligning how our libraries transpile prop types assignment with how our apps already do this.

**Comment by billyjanitsch** [comment]
(Also we already have a [lint rule](https://github.com/yannickcr/eslint-plugin-react/blob/60b5642d0f212e5bfb7eefefc8616ffabdedf566/docs/rules/forbid-foreign-prop-types.md) to guard against foreign prop type access.)

**Comment by jhhayashi** [comment]
this sounds good to me, then. @nabeel- any thoughts?

**Comment by nabeel-** [comment]
Nope, lgtm!

I believe we had a bit of a discussion when we wanted to first introduce this plugin in https://github.com/kensho/babel-preset-kensho/commit/e7de8fc782770e76b23fa46c0c5ce59a286f3a74 and decided to be more relaxed with `wrap`. This seems in line with what we had planned to do and I haven't see anything new crop up that would indicate we should change.

File: test/index.js [hunk 1]
```diff
@@ -34,8 +34,9 @@ test('uses external CJS helpers', snapshot, 'esm-helpers', {modules: 'commonjs',
 test('cherry picks lodash modules', snapshot, 'lodash')
 test('does not cherry pick lodash modules', snapshot, 'lodash', {lodash: false})
 
-test('wraps prop types', snapshot, 'prop-types')
-test('wraps prop types', snapshot, 'prop-types', {removePropTypes: 'wrap'})
+test('wraps prop types assignment', snapshot, 'prop-types')
+test('wraps prop types assignment', snapshot, 'prop-types', {removePropTypes: 'unsafe-wrap'})
+test('wraps prop types value', snapshot, 'prop-types', {removePropTypes: 'wrap'})
```
Comment (nabeel-) [inline test/index.js]: What's the reasoning behind leaving in this test?

Comment (billyjanitsch) [reply test/index.js]: Just in case a consumer wants to manually use that option for a specific reason (maybe orama, to be extra careful?)

Comment (nabeel-) [reply test/index.js]: Cool, it just felt like it's slightly more of a test for the ability to override a value in the config.

Comment (billyjanitsch) [reply test/index.js]: Yeah it is, but the whole suite really just tests that certain config/env values result in expected transpilation output.

**Pull request merged**